doc/userdoc/styleguide/styleguide.rst

```diff
@@ -0,0 +1,566 @@
+
+Use double backticks for
+
+
+  - inline code
+  - objects/functions
+  - model names
+  - NEST-specific vocabulary
+  - function call (e.g., ``nest.Create("iaf_psc_alpha")`` or  (``Create`` )
+  - Paths (e.g, You can find the models in ``nest-simulator/pynest/examples``)
+  - Key value pairs (``{key: value}``)
+  - Variables with assigned values ``x = 10``
+
+
+\`Single backticks\`
+~~~~~~~~~~~~~~~~~~~~
+
+Use single backticks for
+
+- Keys
+- Parameters
+- Variable names
+- Values
+
+but use double backticks when showing a complete example of variable with
+assigned value (e.g., \``volt = 37.0``)
+
+An example:
+
+::
+
+   Here we use the ``Create`` function to instantiate our model, in this case
+   ``iaf_psc_alpha``. We can modify the parameter `V_m` and set the value to
+   `50.0`.
+
+Rendered as
+
+
+   Here we use the ``Create`` function to instantiate our model, in this case
+   ``iaf_psc_alpha``. We can modify the parameters `V_m` and set the value to
+   `50.0`.
+
+\**Boldface\**
+~~~~~~~~~~~~~~
+
+If you want to emphasize a word or phrase in text, you can use **boldface**.
+
+NumPy style docstrings
+~~~~~~~~~~~~~~~~~~~~~~
+
+In the PyNEST code, we follow the rules for NumPy style docstrings as
+`explained here <https://numpydoc.readthedocs.io/en/latest/format.html>`_.
+
+
+reStructuredText markup
+-----------------------
+
+Let us take a brief look into the markup of reStructuredText. You can find
+a full cheatSheet `here <https://thomas-cokelaer.info/tutorials/sphinx/
+rest_syntax.html>`_.
+
+.. note::
+
+   reStructuredText is sensitive to indentation. A single newline should always
+   be between any text and headings, tables, reference labels, or directives.
+   If the output format seems incorrect, double check the spaces and newlines
+   of the text.
+
+Code and code snippets
+~~~~~~~~~~~~~~~~~~~~~~
+
+For writing Python code, you can use the double colon
+
+
+ \::
+
+Rendered as
+
+   ::
+
+       python code here
+
+       x = 1
+
+       for i in line:
+           x = i
+           print(thing.x)
+
+For other code styles use the code-block directive (you can also use this one
+for python)
+
+
+
+\   .. code-block:: cpp
+
+Rendered as
+
+    .. code-block:: cpp
+
+       int main()
+       {
+         cout << "Hello, World!";
+         return 0;
+       }
+
+
+For input/output examples use the following syntax
+
+::
+
+   >>> input begins with 3, right-angled brackets
+   output is directly below input without any brackets
+
+
+Directives
+~~~~~~~~~~~
+
+This is not a complete guide to the directives of reStructuredText. For more
+options see `Sphinx directives <https://www.sphinx-doc.org/en/master/usage/
+restructuredtext/directives.html>`_ or `Docutils section on directives
+<http://docutils.sourceforge.net/docs/ref/rst/directives.html>`_.
+
+Equations
+^^^^^^^^^
+
+For equations, use LaTeX markup:
+
+::
+
+    .. math::
+
+            f(x) = \int_{-\infty}^{\infty} \hat f(\xi)\ e^{2 \pi i x \xi}\,d\xi,
```

Comment:
I am not sure I agree with the LaTeX typesetting here. If I remember my Knuth correctly, it should be
```suggestion
            f(x) = \int_{-\infty}^{\infty} \hat{f}(\xi) \mathrm{e}^{2 \pi i x \xi} \mathrm{d}\xi \;.
```
The logic is as follows:
- LaTeX handles whitespace within math expressions automatically
- Since e and d are no variables, they should be set in roman
- In the example here, the equation is at the end of a sentence and should thus end with a fullstop. The fullstop is set off with a slightly wider space from the equation.

The same changes apply to the "rendered" line below.

Reply:
I copied this from somewhere (Wikipedia?) as my LaTeX equation knowledge is virtually non-existent; so if it renders, I interpret it to be correct.
Do we need some sort of reference here to show how to properly format LaTeX?

Reply:
I modified my suggestion above. There are different schools of thought especially on whether to use roman face for Euler's constant e  and the differential d, and concerning spacing before punctuation. I would suggest the following references for mathematics style:
- [Mathematics into Type](http://www.ams.org/arc/styleguide/mit-2.pdf) by Swanson et al (1999), published by the American Mathematical Society (AMS)
- [AMS Style Guide](http://www.ams.org/publications/authors/AMS-StyleGuide-online.pdf), essentially the same guidelines as in *Mathematics into Type* but even more detailed, aimed at copy editors
- Wikipedia guidelines on [Typesetting of mathematical formulae](https://en.wikipedia.org/wiki/Wikipedia:Manual_of_Style/Mathematics#Typesetting_of_mathematical_formulae)

Those all agree that Euler's `e` and integral's `d` are italicised, that a thin space (`\,`) is to be manually inserted before `dx` and that there is no spacing between and equation and punctuation following it. AMS does *not* insert a space before Euler's `e` in `f(x) e^x`, see the examples in Sec 13.11 of the AMS Style Guide.

Reply:
Fixed, i added only the first refernece in the styleguide as I think that should be enough